src/linter/index.tsx

```diff
@@ -0,0 +1,78 @@
+type Message = {
+  column: number | null;
+  description: string | string[];
+  file: string;
+  line: number | null;
+  message: string;
+  type: 'notice' | 'error' | 'warning';
+  uid: string;
+};
+
+export type ExternalMessage = Message & {
+  // These are some extra properties that we don't need to work with.
+  context: string[];
+  for_appversions: object;
+  id: string[];
+  tier: number;
+};
+
+type ExternalLinterResult = {
+  error: null | string;
+  full_report_url: string;
+  upload: string;
+  url: string;
+  validation: {
+    detected_type: string;
+    ending_tier: number;
+    errors: number;
+    message_tree: object;
+    messages: ExternalMessage[];
+    metadata: object;
+    notices: number;
+    success: boolean;
+    warnings: number;
+  };
+};
+
+const createInternalMessage = (message: ExternalMessage): Message => {
+  return {
+    column: message.column,
+    description: message.description,
+    file: message.file,
+    line: message.line,
+    message: message.message,
+    type: message.type,
+    uid: message.uid,
+  };
+};
+
+type MessageMap = {
+  [relativeFileName: string]: {
```

Comment:
This prop named `relativeFileName` is called `path` everywhere else (because the API names it that way). I'd suggest to use `path` here or to file a new issue to use an explicit and consistent name everywhere.